dataprofiler/profilers/categorical_column_profile.py

```diff
@@ -323,6 +370,113 @@ def _update_stop_condition(self, data: DataFrame):
             self._stopped_at_unique_ratio = merged_unique_ratio
             self._stopped_at_unique_count = merged_unique_count
 
+    @BaseColumnProfiler._timeit(name="categories")
+    def _get_categories_cms(self, df_series, len_df):
+        """Return count min sketch and heavy hitters for both the batch and stream case.
+
+        :param df_series: Series currently being processed by categorical profiler
+        :type df_series: Series
+        :return: cms, heavy_hitter_dict, missing_heavy_hitter_dict
+        """
+        cms = datasketches.count_min_sketch(self.cms_num_hashes, self.cms_num_buckets)
+        heavy_hitter_dict = defaultdict(int)
+        missing_heavy_hitter_dict = defaultdict(int)
+        for i, value in enumerate(df_series):
+            cms.update(value)
+            i_count = cms.get_estimate(value)
+            i_total_count = i_count + self.cms.get_estimate(value)
+            # approximate heavy-hitters
+            if i_count >= int(len_df / self._cms_max_num_heavy_hitters):
+                heavy_hitter_dict[value] = i_count
+                # missing_heavy_hitter_dict.pop(value, None)
```

Comment:
I think this is problematic if there's a case where it ends in this `if` but gets populated in the second and then gets overwritten. Need to still read below to see if this is problematic.

was the a purpose of removing

Reply:
I created this test case which will fail:
```python
dataset = pd.Series(["b"] * 2 + ["c"] * 14)
dataset1 = pd.Series(["b"] * 5 + ["c"] * 9)

options = CategoricalOptions()
options.cms = True
options.cms_confidence = 0.95
options.cms_relative_error = 0.01
options.cms_max_num_heavy_hitters = 3

profile = CategoricalColumn("test_name", options)
profile.update(dataset)

expected_categories = ["c"]
expected_categories_dict = {"c": 14}

self.assertEqual(profile.sample_size, len(dataset))
self.assertEqual(profile._categories, expected_categories_dict)
self.assertCountEqual(expected_categories, profile.categories)

profile.update(dataset1)
expected_categories = ["c"]
expected_categories_dict = {"c": 23}

self.assertEqual(profile.sample_size, len(dataset) + len(dataset1))
self.assertEqual(profile._categories, expected_categories_dict)
self.assertCountEqual(expected_categories, profile.categories)
```

Reply:
Ultimately, the `update` overwrites:
`AssertionError: defaultdict(<class 'int'>, {'c': 18.0}) != {'c': 24}`

Reply:
uncommenting both of these will pass this test.

Reply:
We can add this test, or find a way to have this be a condition in the current test with a third update.

Reply:
comments were an accident while debugging.

Reply:
adding this test